### PR TITLE
🏗 Lint `OWNERS` files with `gulp lint`

### PR DESCRIPTION
**Pull request by rsimha**
This PR makes it so that `gulp lint`, `gulp lint --local_changes`, and `gulp lint --files=<glob>` are all capable of linting `OWNERS` files.

**Example usage:**
```
$ gulp lint --files=build-system/**/OWNERS
[16:47:00] Using gulpfile ~/src/amphtml/gulpfile.js
[16:47:00] Starting 'lint'...
[16:47:00] All packages in node_modules are up to date.
[16:47:00] INFO: Running lint on the following files:
[16:47:00] build-system/global-configs/OWNERS
[16:47:00] build-system/OWNERS
[16:47:01] SUCCESS: No linter warnings or errors.
[16:47:01] Finished 'lint' after 816 ms
```
```
$ gulp lint --local_changes
[16:47:45] Using gulpfile ~/src/amphtml/gulpfile.js
[16:47:45] Starting 'lint'...
[16:47:45] All packages in node_modules are up to date.
[16:47:45] INFO: Running lint on the following files:
[16:47:45] build-system/OWNERS
[16:47:45] build-system/tasks/lint.js
[16:47:46] SUCCESS: No linter warnings or errors.
[16:47:46] Finished 'lint' after 1 s
```

Follow up to #24938

### CODE REVIEW

File: build-system/tasks/lint.js [hunk 1]
```diff
@@ -161,13 +162,16 @@ function runLinter(stream, options) {
...
-function jsFilesChanged() {
+function lintableFilesChanged() {
   return gitDiffNameOnlyMaster().filter(function(file) {
-    return fs.existsSync(file) && path.extname(file) == '.js';
+    return (
+      fs.existsSync(file) &&
+      (path.extname(file) == '.js' || path.basename(file) == 'OWNERS')
+    );
   });
 }
```
Comment (estherkim) [inline build-system/tasks/lint.js]: Shouldn't we be using `lintGlobs` in `build-system/test-configs/config.js`?

**Comment by estherkim** [comment]
Dang 2 min too late.

File: build-system/tasks/lint.js (hunk 1 above)
Comment (rsimha) [reply build-system/tasks/lint.js]: This was needed in the past when we listed individual files (including negative globs) in `lintGlobs`. Now, we only list two globs in `lintGlobs`: `**/*.js` and `**/OWNERS`. All other negative globs are listed in the global `.eslintignore` (in order to work with IDE plugins). So, when `--files` or `--local_changes` are used, we needn't bother looking at `lintGlobs`.

Comment (estherkim) [reply build-system/tasks/lint.js]: It feels like we're duplicating logic here, though. I get that this function is used only when using these flags, but IMO a cleaner way would be to glob match the files with `lintGlobs`. Or else we'd have to remember to update this function whenever `lintGlobs` is updated.

Comment (rsimha) [reply build-system/tasks/lint.js]: Aha, I now see what you're saying. Agreed. Will do in a follow up.

Comment (rsimha) [reply build-system/tasks/lint.js]: Sent out #24946. Thanks for keeping me honest!

Comment (estherkim) [reply build-system/tasks/lint.js]: :)

**Pull request merged**